src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -388,14 +388,22 @@ private void AddEvaluationActivityEvent(EvaluationEvent evaluationEvent)
 
             ActivityTagsCollection tags = new ActivityTagsCollection
             {
-                { "TargetingId", evaluationEvent.TargetingContext?.UserId },
                 { "FeatureName", evaluationEvent.FeatureDefinition.Name },
-                { "Variant", evaluationEvent.Variant?.Name },
                 { "Enabled", evaluationEvent.Enabled },
                 { "VariantAssignmentReason", evaluationEvent.VariantAssignmentReason },
                 { "Version", ActivitySource.Version }
             };
 
+            if (evaluationEvent.TargetingContext?.UserId != null)
+            {
+                tags["TargetingId"] = evaluationEvent.TargetingContext.UserId;
+            }
+
+            if (evaluationEvent.Variant?.Name != null)
+            {
+                tags["Variant"] = evaluationEvent.Variant.Name;
+            }
+
```

Comment:
I'd suggest using pattern matching for these as well:
```suggestion
            if (evaluationEvent.TargetingContext is { UserId: string userId })
            {
                tags["TargetingId"] = userId;
            }

            if (evaluationEvent.Variant is { Name: string variantName })
            {
                tags["Variant"] = variantName;
            }

```